### PR TITLE
[MLOps 1.5] [PrometheusLogger] Refactor naming convention

### PR DESCRIPTION
**Pull request by dbogunowicz**
This pull request updates the metric naming scheme in Prometheus to conform to best practices by adding labels, a prefix, and suffixes. 

The following changes are made:

- A "deepsparse_" prefix is added to every Prometheus metric to represent the namespace.
- Prediction latency metrics have a "_seconds" suffix.
- Memory available - "_bytes"
- Cores used - "_total"
- Other identifiers are also adjusted accordingly
- Adding `model_name` as a label is a bit tricky. It has been implemented separately here: #884 

## Examples of new naming convention in Prometheus
(please disregard the erroneous attribution of `model_name`, to be fixed)
### `config.yaml`
```bash
loggers:
    prometheus:
      port: 8001
system_logging:
    prediction_latency:
        enable: true
    resource_utilization:
        enable: true
    request_details:
        enable: true
num_cores: 1
num_workers: 1
endpoints:
    - task: question_answering
      model: zoo:nlp/question_answering/distilbert-none/pytorch/huggingface/squad/pruned80_quant-none-vnni
      data_logging:
        pipeline_outputs.score:
          - func: identity
```
### logging `num_cores`
```bash
# HELP deepsparse_question_answering__0__resource_utilization__num_cores_total summary metric for identifier: question_answering-0/resource_utilization/num_cores_total | Category: MetricCategories.SYSTEM
# TYPE deepsparse_question_answering__0__resource_utilization__num_cores_total summary
deepsparse_question_answering__0__resource_utilization__num_cores_total_count{model_name="num_cores_total"} 1.0
deepsparse_question_answering__0__resource_utilization__num_cores_total_sum{model_name="num_cores_total"} 1.0
# HELP deepsparse_question_answering__0__resource_utilization__num_cores_total_created summary metric for identifier: question_answering-0/resource_utilization/num_cores_total | Category: MetricCategories.SYSTEM
# TYPE deepsparse_question_answering__0__resource_utilization__num_cores_total_created gauge
deepsparse_question_answering__0__resource_utilization__num_cores_total_created{model_name="num_cores_total"} 1.6745612586558185e+09
```

### logging `prediction_latency`
```bash
# HELP deepsparse_question_answering__0__prediction_latency__pre_process_seconds summary metric for identifier: question_answering-0/prediction_latency/pre_process_seconds | Category: MetricCategories.SYSTEM
# TYPE deepsparse_question_answering__0__prediction_latency__pre_process_seconds summary
deepsparse_question_answering__0__prediction_latency__pre_process_seconds_count{model_name="pre_process_seconds"} 3.0
deepsparse_question_answering__0__prediction_latency__pre_process_seconds_sum{model_name="pre_process_seconds"} 0.00790767977014184
# HELP deepsparse_question_answering__0__prediction_latency__pre_process_seconds_created summary metric for identifier: question_answering-0/prediction_latency/pre_process_seconds | Category: MetricCategories.SYSTEM
# TYPE deepsparse_question_answering__0__prediction_latency__pre_process_seconds_created gauge
deepsparse_question_answering__0__prediction_latency__pre_process_seconds_created{model_name="pre_process_seconds"} 1.6745612615956821e+09
```
### logging `cpu_utilization`
```bash
# HELP deepsparse_resource_utilization__cpu_utilization_percent summary metric for identifier: resource_utilization/cpu_utilization_percent | Category: MetricCategories.SYSTEM
# TYPE deepsparse_resource_utilization__cpu_utilization_percent summary
deepsparse_resource_utilization__cpu_utilization_percent_count{model_name="cpu_utilization_percent"} 3.0
deepsparse_resource_utilization__cpu_utilization_percent_sum{model_name="cpu_utilization_percent"} 0.0
# HELP deepsparse_resource_utilization__cpu_utilization_percent_created summary metric for identifier: resource_utilization/cpu_utilization_percent | Category: MetricCategories.SYSTEM
# TYPE deepsparse_resource_utilization__cpu_utilization_percent_created gauge
deepsparse_resource_utilization__cpu_utilization_percent_created{model_name="cpu_utilization_percent"} 1.6745612616460366e+09
```

### logging `pipeline_outputs.score`
```bash
# HELP deepsparse_question_answering__0__pipeline_outputs__score__identity summary metric for identifier: question_answering-0/pipeline_outputs.score__identity | Category: MetricCategories.DATA
# TYPE deepsparse_question_answering__0__pipeline_outputs__score__identity summary
deepsparse_question_answering__0__pipeline_outputs__score__identity_count{model_name="pipeline_outputs.score__identity"} 3.0
deepsparse_question_answering__0__pipeline_outputs__score__identity_sum{model_name="pipeline_outputs.score__identity"} -3.8007187843322754
# HELP deepsparse_question_answering__0__pipeline_outputs__score__identity_created summary metric for identifier: question_answering-0/pipeline_outputs.score__identity | Category: MetricCategories.DATA
# TYPE deepsparse_question_answering__0__pipeline_outputs__score__identity_created gauge
deepsparse_question_answering__0__pipeline_outputs__score__identity_created{model_name="pipeline_outputs.score__identity"} 1.6745623245058227e+09
```

### CODE REVIEW

File: src/deepsparse/loggers/prometheus_logger.py [hunk 1]
```diff
@@ -97,8 +98,9 @@ def log(self, identifier: str, value: Any, category: MetricCategories):
         :param value: The data structure that the logger is logging
         :param category: The metric category that the log belongs to
         """
+        model_name = identifier.split("/")[-1]
```
Comment (dbogunowicz) [inline src/deepsparse/loggers/prometheus_logger.py:101]: This is not handling all the cases, let's sync on that.

Comment (dbogunowicz) [reply src/deepsparse/loggers/prometheus_logger.py:103]: This is being done in a separate PR: #884

**Review by bfineran — DISMISSED** [review]

File: src/deepsparse/loggers/prometheus_logger.py [hunk 2]
```diff
@@ -168,19 +174,20 @@ def get_prometheus_metric(
...
+def format_identifier(identifier: str, namespace: str = _NAMESPACE) -> str:
     """
     Replace forbidden characters with `__` so that the identifier
     digested by prometheus adheres to
     https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
     :param identifier: The identifier to be formatted
     :return: The formatted identifier
     """
-    return re.sub(r"[^a-zA-Z0-9_]", "__", identifier).lower()
+    return f"{namespace}_{re.sub(r'[^a-zA-Z0-9_]', '__', identifier).lower()}"
```
Comment (bfineran) [inline src/deepsparse/loggers/prometheus_logger.py:190]: nice to leave a comment describing what a regex pattern does (or is intended to do :) )

Comment (dbogunowicz) [reply src/deepsparse/loggers/prometheus_logger.py:190]: I think that the docstrings are quite explicit no? When the user opens the link, they can instantaneously see the same regex there: `[a-zA-Z_:][a-zA-Z0-9_:]*` (actually colon is not really allowed)

**Review by bfineran — APPROVED** [review]

**Review by corey-nm — APPROVED** [review]

**Pull request merged**